vault/identity_store_oidc_provider.go

```diff
@@ -161,3 +216,129 @@ func (i *IdentityStore) pathOIDCAssignmentExistenceCheck(ctx context.Context, re
 
 	return entry != nil, nil
 }
+
+// pathOIDCCreateUpdateScope is used to create a new scope or update an existing one
+func (i *IdentityStore) pathOIDCCreateUpdateScope(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	name := d.Get("name").(string)
```

Comment:
We should return an error if a user tries to create the `openid` scope. The `openid` scope is reserved and maps to a specific set of reserved claims.